File: mobile/lib/features/album/domain/entities/album.dart
class Album {
  final String id;
  final String userId;
  final String name;
  final String? coverPhotoId;
  final bool isPublic;
  final String? shareToken;
  final DateTime createdAt;
  final DateTime updatedAt;

  const Album({
    required this.id,
    required this.userId,
    required this.name,
    this.coverPhotoId,
    this.isPublic = false,
    this.shareToken,
    required this.createdAt,
    required this.updatedAt,
  });
}

File: web/src/lib/types/album.ts
import type { Photo } from "./photo";

export interface Album {
  id: string;
  user_id: string;
  name: string;
  cover_photo_id: string | null;
  is_public: boolean;
  share_token: string | null;
  photo_count: number;
  created_at: string;
  updated_at: string;
}

export interface AlbumDetail extends Album {
  photos: Photo[];
}

export interface AlbumCreate {
  name: string;
  cover_photo_id?: string;
  is_public?: boolean;
}

export interface AlbumUpdate {
  name?: string;
  cover_photo_id?: string;
  is_public?: boolean;
}

export interface AlbumAddPhotos {
  photo_ids: string[];
}
